src/vm/tests/test-docker.js

```diff
@@ -1121,3 +1123,194 @@ test('test docker VM with good zlog_max_size', function (t) {
         });
     }]);
 });
+
+/*
+ * Tries to determine if the array of (Number) values is generally going up or
+ * down. Returns -1 if more downs than ups, 1 if more ups than downs and 0 if
+ * there are the same number of ups and downs or the values are all equal.
+ */
+function trend(values)
+{
+    var downs = 0;
+    var prev_value = values[0];
+    var ups = 0;
+
+    values.forEach(function (val) {
+        if (val > prev_value) {
+            ups++;
+        } else if (val < prev_value) {
+            downs++;
+        }
+        prev_value = val;
+    });
+
+    if (ups === downs) {
+        return (0);
+    } else if (ups > downs) {
+        return (1);
+    } else {
+        return (-1);
+    }
+}
+
+/*
+ * This tests that OS-4740 is still fixed. Specifically if a docker container
+ * has trouble restarting the delay should be increasing, but if it stays up for
+ * at least 10 seconds the delay should get reset.
+ *
+ * This relies on vmadmd being running (since that's what handles restarts)
+ */
+test('test restart delay reset', function (t) {
+    var cycles_fail = 7;
+    var cycle_reset_delay = 15; // seconds, should be >= 10
+    var emitter = new EventEmitter();
+    var events = [];
+    var payload = JSON.parse(JSON.stringify(common_payload));
+    var se;
+    var state = {brand: payload.brand};
+    var num_cycles = (cycles_fail * 2) + 1;
+
+    payload.autoboot = false;
+    payload.archive_on_delete = true;
+    payload.brand = 'lx';
+    payload.docker = true;
+    payload.image_uuid = vmtest.CURRENT_DOCKER_ALPINE_UUID;
+    // This cmd will 'exit 1' cycles_fail times, then sleep cycle_reset_delay
+    // seconds and exit 0, then repeat that pattern.
```

Comment:
it repeats the pattern because it deletes the files in /var/tmp with: `rm -f /var/tmp/*` at which point the `$(ls -1 /var/tmp | wc -l)` will go back to 0.